server/v2/core/store/types.go

```diff
@@ -40,6 +40,7 @@ type WritableState interface {
 	Delete(key []byte) error
 	ApplyChangeSets(changes []ChangeSet) error
 	ChangeSets() ([]ChangeSet, error)
+	WorkingHash() ([32]byte, error)
```

Comment:
it's not good if `WorkingHash` leaks into `branchdb` as seen here: https://github.com/cosmos/cosmos-sdk/pull/18924/files#diff-df719bfa4a37c456b339aa2c62c2b89cf43153f83de75889eeb2ca0eb5fcb0cdR82

because it ties the branch store implementation to the underlying commitment structure, which is not the case since ideallly branch DB is the fastest version of an in-memory DB (with transactional capabilities), independent of the commitment structure

Reply:
Note, `WorkingHash()` will need to be called ultimately during `FinalizeBlock`-equivalent method. That. method is exposed on `RootStore` in store v2.